### ISSUE TITLE
Inconsistent Menu Entry Swapper Behavior for Portal Nexus and Jewellery Box

### ISSUE THREAD

**Issue by F-Lambda** [runelite/runelite#15548]
### Type

Other

### Operating System

Windows

### OS Version

Windows 10 Version 21H2 (OS Build 19044.1947)

### Bug description

Both the Portal Nexus and the Jewellery Box in the player-owned house have inconsistent behavior when using the Menu Entry Swapper plugin. Ideally, you'd be able to customize these house furniture to allow you to either bring up the Teleport Menu or go directly to the Previous Destination, depending on whether you Click or Shift-Click the furniture. However, current implementation of menu-entry swapper does not allow this.

The default configuration without the plugin is Teleport Menu for both Click and Shift-Click, with Previous Destination available on right-click.
[![image](https://user-images.githubusercontent.com/21232805/186323784-60c5f547-035c-43b6-aae4-e55d6175bada.png)](https://i.imgur.com/sduZnD1.png)

Shift-Right Clicking allows you to set the Click or Shift-Click to Previous Destination. However, this is only saved for that specific destination, requiring you to need to set it for every destination individually if you want consistent behavior. For example, You can swap Grand Exchange to be the Click on the Jewellery Box, but if you then teleport to the Woodcutting Guild, then that will only be available via Right-Click.
[![image](https://user-images.githubusercontent.com/21232805/186323523-3d954c08-1794-440b-8a73-5b76d9d46c50.png)](https://i.imgur.com/lrTpfvf.png)

In the Menu Entry Swapper plugin settings, you can click a checkbox to universally swap Teleport Menu with Previous Destination. Like default, with this option enabled Previous Destination is both Click and Shift-Click, with Teleport Menu hidden behind a Right-Click.
[![image](https://user-images.githubusercontent.com/21232805/186323276-feb7d7ee-a4cd-4751-8659-61918ad5b0f5.png)](https://i.imgur.com/juUcHiW.png)

When you try to Shift-Right Click to custom swap entries, you find that the only options are to set Previous Destination as Click or Shift-Click, but this is already the case.
[![image](https://user-images.githubusercontent.com/21232805/186324003-ca448175-60b1-4a8b-938d-234e940d43fb.png)](https://i.imgur.com/grj85NK.png)

Thus, in the current state, there is no way to set Jewellery Boxes or Portal Nexus to consistently have Click be Previous Destination and Shift-Click be Teleport Menu, or vice versa. Leaving the settings menu option unchecked requires custom setting Click for every option, while checking it doesn't allow you to set Teleport Menu to Shift-Click at all.

### RuneLite version

RuneLite version: 1.8.31
Oldschool version: Rev 207
Launcher version: 2.4.2

**Comment by F-Lambda** [comment]
With the push in recent updates to move towards shift click customization for swapping instead of menu settings, the best long term solution would probably be to eliminate the menu setting while making sure all options are actually presented when shift clicking, and to customize for all previous destination options when one is picked, and not just that specific one.

**Comment by geheur** [comment]
Fairy rings have the same issue when using the custom swapper. Thankfully there is a fairy ring specific swap but iirc it's for left click only.

**Comment by F-Lambda** [comment]
> Fairy rings have the same issue when using the custom swapper. Thankfully there is a fairy ring specific swap but iirc it's for left click only.

Yeah, I noticed that as well. Looking to see if it was already mentioned, I saw in #14864 that it gets even more complicated with the Tree & Fairy ring, in that you have to set the Fairy Ring default to Zanaris in order to have the Spirit Tree not be hidden behind right click

**Comment by m00fin** [comment]
Enabling 'Shift click walk here' causes same bug as if the 'Jewellery Box' option has been enabled, it becomes impossible to assign 'Teleport Menu' to shift-click. Seems related so thought I'd add here instead of a new issue.
